selfdrive/ui/qt/offroad/networking.cc

```diff
@@ -43,29 +44,39 @@ std::string exec(const char* cmd) {
 
 // Networking functions
 
-Networking::Networking(QWidget* parent, bool show_advanced) : QWidget(parent){
+Networking::Networking(QWidget* parent, bool show_advanced) : QWidget(parent), show_advanced(show_advanced){
+  s = new QStackedLayout;
+
+  QLabel* warning = new QLabel("Network manager is inactive!");
+  warning->setStyleSheet(R"(font-size: 65px;)");
+
+  s->addWidget(warning);
+  setLayout(s);
+
+  startup_timer = new QTimer(this);
+  connect(startup_timer, SIGNAL(timeout()), this, SLOT(attemptInitialization()));
+  startup_timer->start(1000);
+}
+
+bool Networking::attemptInitialization(){
+  // Checks if network manager is active
   try {
     wifi = new WifiManager(this);
   } catch (std::exception &e) {
-    QLabel* warning = new QLabel("Network manager is inactive!");
-    warning->setStyleSheet(R"(font-size: 65px;)");
-
-    QVBoxLayout* warning_layout = new QVBoxLayout;
-    warning_layout->addWidget(warning, 0, Qt::AlignCenter);
-    setLayout(warning_layout);
-    return;
+    delete(wifi);
+    return false;
   }
+
   connect(wifi, SIGNAL(wrongPassword(QString)), this, SLOT(wrongPassword(QString)));
 
-  s = new QStackedLayout;
 
   QVBoxLayout* vlayout = new QVBoxLayout;
 
   if (show_advanced) {
     QPushButton* advancedSettings = new QPushButton("Advanced");
     advancedSettings->setStyleSheet(R"(margin-right: 30px)");
-    advancedSettings->setFixedSize(300, 100);
-    connect(advancedSettings, &QPushButton::released, [=](){s->setCurrentIndex(1);});
+    advancedSettings->setFixedSize(350, 100);
+    connect(advancedSettings, &QPushButton::released, [=](){s->setCurrentIndex(2);});
```

Comment:
use setCurrentWidget for cases like this, it's much easier to read and robust to changes like this